modules/xfem/src/base/XFEM.C

```diff
@@ -882,6 +929,148 @@ XFEM::initCutIntersectionEdge(
   return does_intersect;
 }
 
+bool
+XFEM::healMesh()
+{
+  bool mesh_changed = false;
+
+  std::set<Node *> nodes_to_delete;
+  std::set<Node *> nodes_to_delete_displaced;
+  std::set<unsigned int> cutelems_to_delete;
+
+  for (unsigned int i = 0; i < _geometric_cuts.size(); ++i)
+  {
+    if (_geometric_cuts[i]->healMesh())
+    {
+      for (ElementPairLocator::ElementPairList::iterator it =
+               _sibling_elems[_geometric_cuts[i]->getInterfaceID()].begin();
+           it != _sibling_elems[_geometric_cuts[i]->getInterfaceID()].end();
+           ++it)
+      {
+        Elem * elem1 = const_cast<Elem *>(it->first);
+        Elem * elem2 = const_cast<Elem *>(it->second);
```

Comment:
A const_cast? Perhaps you shouldn't store these Elems as const if you are just going to cast it away. It might will cause confusion since developers might think you are actually not modifying those elements when they see the declarations.

Reply:
For most of other cases, we still want these elements to be const.